### PR TITLE
feat(vps.snapshot): add download vps snapshot feature

### PR DESCRIPTION
**Pull request by brofik**
| Question         | Answer
| ---------------- | ---
| Branch?          | `release/component-w11` 
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | yes
| Tickets          | Fix #MANAGER-8766 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description
Created because https://github.com/ovh/manager/pull/8530 is close because of `stale` status.

### CODE REVIEW

**Comment by sachinrameshn** [comment]
Can you please add correct ticket ref in commit message ?

**Review by ghyenne — APPROVED** [review]

File: packages/manager/modules/vps/src/vps/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -99,6 +99,16 @@
...
   "vps_dashboard_monitoring_sla_ips_tooltip": "Pour le bon fonctionnement de nos outils de monitoring, merci d'autoriser ces plages d'IPs dans votre firewall.",
+  "vps_dashboard_snapshot_download_modal_title": "Télécharger Snapshot",
+  "vps_dashboard_snapshot_download_modal_summary": "Informations",
+  "vps_dashboard_snapshot_download_modal_service_name": "Nom du service",
+  "vps_dashboard_snapshot_download_modal_snapshot_size": "Taille",
+  "vps_dashboard_snapshot_download_modal_snapshot_creation_date": "Date de création du snapshot",
+  "vps_dashboard_snapshot_download_modal_snapshot_link_expiration_date": "Date d'expiration du lien de téléchargement",
+  "vps_dashboard_snapshot_download_modal_snapshot_loading_message": "Génération de la commande de téléchargement du snapshot ...",
+  "vps_dashboard_snapshot_download_modal_snapshot_curl_success": "Commande de téléchargement générée avec succès.",
+  "vps_dashboard_snapshot_download_modal_snapshot_curl_info": "Veuillez copier la commande CURL ci-dessus et l'exécuter sur votre terminal. Pour en savoir plus sur CURL <a href=\"{{curlDocumentationLink}}\" target=\"_blank\">Cliquez ici</a>.",
```
Comment (anooparveti) [inline packages/manager/modules/vps/src/vps/translations/Messages_fr_FR.json:110]: **suggestion:**  `Cliquez ici` can be separated into a different sentence and avoid HTML in Translation files

File: packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  $onInit() {
+    this.size = null;
+    this.creationDate = moment(this.tabSummary.snapshot.creationDate).format(
+      'LLL',
+    );
+    this.linkExpirationDate = this.getLinkExpirationDate();
+    this.downloadLink = null;
+    this.downloadLinkCurlCommand = null;
```
Comment (anooparveti) [inline packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js:19]: **nitpick:** unnecessary initialisation to `null`
```suggestion
```

File: packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+export default class SnapshotDownloadController {
+  /* @ngInject */
+  constructor($translate, atInternet, VpsService) {
+    this.$translate = $translate;
+    this.atInternet = atInternet;
+    this.VpsService = VpsService;
+  }
+
+  $onInit() {
+    this.size = null;
```
Comment (anooparveti) [inline packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js:13]: **nitpick:** unnecessary initialisation to `null`
```suggestion
```

File: packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    this.creationDate = moment(this.tabSummary.snapshot.creationDate).format(
+      'LLL',
+    );
+    this.linkExpirationDate = this.getLinkExpirationDate();
+    this.downloadLink = null;
+    this.downloadLinkCurlCommand = null;
+    this.isLoading = false;
+    this.notificationMessage = null;
+    this.errorMessage = null;
```
Comment (anooparveti) [inline packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js:22]: **nitpick:** unnecessary initialisation to `null`
```suggestion
```

File: packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        break;
+      case this.status.SUCCESS:
+        this.notificationMessage =
+          'vps_dashboard_snapshot_download_modal_snapshot_curl_success';
+        break;
+      default:
+        this.notificationMessage = null;
+    }
+    return this.notificationMessage;
+  }
```
Comment (y4nnL) [inline packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js:107]: Get methods shouldn't have assign statements

File: packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.html [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        </div>
+    </div>
+
+    <div
+        class="d-flex justify-content-center mt-3"
+        data-ng-if="!$ctrl.downloadLink"
+        data-ng-hide="$ctrl.isLoading"
+    >
+        <oui-button
+            variant="primary"
```
Comment (y4nnL) [inline packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.html:81]:
```suggestion
            data-variant="primary"
```

**Review by anooparveti — APPROVED** [review]

**Pull request merged**